storage/redis_cluster.go

```diff
@@ -497,157 +309,61 @@ func (r *RedisCluster) GetKeysAndValues() map[string]string {
 
 // DeleteKey will remove a key from the database
 func (r *RedisCluster) DeleteKey(keyName string) bool {
-	if err := r.up(); err != nil {
-		log.Debug(err)
-		return false
-	}
-	log.Debug("DEL Key was: ", keyName)
-	log.Debug("DEL Key became: ", r.fixKey(keyName))
-	n, err := r.singleton().Del(r.RedisController.ctx, r.fixKey(keyName)).Result()
-	if err != nil {
-		log.WithError(err).Error("Error trying to delete key")
-	}
-
-	return n > 0
+	return r.DeleteRawKey(r.fixKey(keyName))
 }
 
 // DeleteAllKeys will remove all keys from the database.
 func (r *RedisCluster) DeleteAllKeys() bool {
 	if err := r.up(); err != nil {
-		log.Debug(err)
 		return false
 	}
-	n, err := r.singleton().FlushAll(r.RedisController.ctx).Result()
-	if err != nil {
-		log.WithError(err).Error("Error trying to delete keys")
-	}
 
-	if n == "OK" {
-		return true
-	}
+	n, err := r.singleton().FlushAll(r.context())
+	r.check(err, "DeleteAllKeys")
 
-	return false
+	return n
 }
 
-// DeleteKey will remove a key from the database without prefixing, assumes user knows what they are doing
+// DeleteKey will remove a key from the database without prefixing
 func (r *RedisCluster) DeleteRawKey(keyName string) bool {
 	if err := r.up(); err != nil {
-		log.Debug(err)
 		return false
 	}
-	n, err := r.singleton().Del(r.RedisController.ctx, keyName).Result()
-	if err != nil {
-		log.WithError(err).Error("Error trying to delete key")
-	}
 
-	return n > 0
+	err := r.singleton().Del(r.context(), keyName)
+	r.check(err, "DeleteRawKey", keyName)
+
+	return err == nil
```

Comment:
before we were returning true when` n>0` (I guess when something was deleted). Now when `err == nil`. Is it the same underline?

Reply:
I don't think we care if the key existed before. As long as there is no error, we can asume the result is one of the following: key does not exist, or key is deleted. Usage checks here are rare, so for the most part gateway doesn't check the output in a fire-and-forget fashion.